### PR TITLE
Add default values to usage

### PR DESCRIPTION
**Pull request by walle**
Check if the value isn't it's zero value and if not add a default value
to the usage text.

Updated implementation of default values in usage from discussion in #18

### CODE REVIEW

**Comment by kentquirk** [comment]
It's up to @alexflint to decide if the format is right. Personally, I'd have put it in parentheses. But whatever. I like this change.

**Comment by walle** [comment]
Good input @kentquirk I'm going to add parentheses around the default value, looks nicer and you see that it's a special value.

**Comment by brettlangdon** [comment]
I think the general convention is to use brackets not parenthesis, see [docopt](http://docopt.org/)

But again, this is probably for @alexflint to decide.

**Comment by walle** [comment]
Good point @brettlangdon, I've updated to brackets.

File: usage.go [hunk 1]
```diff
@@ -95,6 +95,13 @@ func (p *Parser) WriteHelp(w io.Writer) {
 				}
 				fmt.Fprint(w, spec.help)
 			}
+			// Check if spec.dest is zero value or not
+			// If it isn't a default value have been added
+			v := spec.dest
+			z := reflect.Zero(v.Type())
+			if v.Interface() != z.Interface() {
+				fmt.Fprintf(w, " [default: %s]", v)
```
Comment (alexflint) [inline usage.go]: For non-strings, I believe this will end up printing something like `%!s(reflect.Value=1)`. Can you use `%q` instead?

**Comment by alexflint** [comment]
This looks good. Sorry for delay in responding (I'm actually getting married this weekend, so have been a bit busy!) One comment then happy to merge.

File: usage.go (hunk 1 above)
Comment (walle) [inline usage.go]: Changed this to use %v instead. %q did not work for int (I added an int value to the usage test to verify that it works for non string values) it gave '*'

**Comment by walle** [comment]
I have updated the code (and the test) to use %v instead of %s (see my line comment). 

Congratulations on getting married!

**Comment by alexflint** [comment]
Thanks @walle. After conflicts are resolved I believe this will be good to go.

**Comment by walle** [comment]
I've rebased and fixed the merge conflicts. I think I got it right, but would like another eye on it. The tests pass though.

Since the new code shows built ins, that don't read from spec.dest but spec.isBool, an extra check is needed to not try to read from the zero value of spec.dest. https://github.com/alexflint/go-arg/pull/19/files#diff-3472cb4e0a973362a7dbc843481a069eR107

**Comment by alexflint** [comment]
Thanks @walle!

**Pull request merged**